core/domain/exp_services.py

```diff
@@ -1960,6 +1960,8 @@ def get_exp_with_draft_applied(exp_id, user_id):
     """
     exp_user_data = user_models.ExplorationUserDataModel.get(user_id, exp_id)
     exploration = exp_fetchers.get_exploration_by_id(exp_id)
+    draft_change_list = None
+    draft_change_list_exp_version = None
```

Comment:
I added these lines over here just to change the scope of these two variables since we are using these later on inside `are_changes_mergeable`.
Refer to line 1989 of this file.
![image](https://user-images.githubusercontent.com/75239620/156404902-6c451601-b23f-48cf-ac04-3d95c5cb19ce.png)

As you can see inside the screenshot, `draft_change_list` and `draft_change_list_exp_version` are defined under `if` statement (initially) so we can not use them outside that particular `if` statement.

Reply:
Hmm, I think it is fine, this whole function would probably be better refactored though, but that is a bit out of scope here.

Reply:
Can you add a TODO here?

Reply:
Added.